### PR TITLE
mask hotpixels in a spectrum

### PR DESCRIPTION
**Pull request by YuiKasagi**
I created the function `corr_hotpix`. This can remove hotpixels in the 1D spectrum by using a 2D hotpixel mask.
(Previously, a hotpixel mask was used only when removing the RN pattern, not for an extracted spectrum.)
Note that the hotpixel mask used may need to be upgraded as issued #8 (I will work on this next.)

### CODE REVIEW

File: src/pyird/utils/irdstream.py [hunk 1]
```diff
@@ -229,10 +230,65 @@ def flatten(self, trace_path=None, extout='_fl', extin=None, mask=None):
...
                 hdulist.writeto(save_path, overwrite=True)
+
+        if self.info:
+            print('flatten (+ hotpix mask): output extension=', extout)
+
         self.fitsdir = self.anadir
         self.extension = extout
         os.chdir(currentdir)
 
+    def corr_hotpix(self,hotpix_mask, rsd, y0, xmin, xmax, coeff):
```
Comment (HajimeKawahara) [inline src/pyird/utils/irdstream.py:241]: `corr` can be interpreted as `correct` and `correlated` etc?

Comment (HajimeKawahara) [reply src/pyird/utils/irdstream.py:241]: e.g. `apply_hotpixel_mask` is more direct?

**Review by HajimeKawahara — APPROVED** [review]
Thanks! All of the unit tests passed.

File: src/pyird/utils/irdstream.py (hunk 1 above)
Comment (YuiKasagi) [reply src/pyird/utils/irdstream.py:241]: That's true. I will change it to `apply_hotpixel_mask`.

**Review by kemasuda — APPROVED** [review]
All the unit tests and irdstream.py ran successfully in my environment.

File: src/pyird/utils/irdstream.py (hunk 1 above)
Comment (HajimeKawahara) [reply src/pyird/utils/irdstream.py]: thanks!

**Pull request merged**